pkg/kube/ready.go

```diff
@@ -131,7 +135,13 @@ func (c *ReadyChecker) IsReady(ctx context.Context, v *resource.Info) (bool, err
 		if err != nil {
 			return false, err
 		}
-		if !c.volumeReady(claim) {
+
+		storageClass, err := getStorageClassFromPvc(ctx, c.client, claim)
```

Comment:
If I were to nitpick, perhaps `volumeReady` should (continue) to have all "volume ready" logic. And should call `getStorageClassFromPvc `.

(perhaps this would make `volumeReady` a candidate for refactoring itself, since adding this call would increase complexity there; but I don't think the right approach is to put some of the functionality into the caller here)